### PR TITLE
feat: multichain popular network dapp connection confirmation bottomsheet

### PR DESCRIPTION
**Pull request by EtherWizard33**
## **Description**
It's rather simple all it does is modify the content of an existing screen.
And it does so behind a feature flag, meaning it keeps the previous UI when the flag is false, and shows the modified version if the flag is true.

## **Related issues**

Contributes to solve issue https://github.com/MetaMask/MetaMask-planning/issues/2803

## **Manual testing steps**

1. from a dapp tab, in the dapp, switch network to polygon
2. the bottom sheet seen below will show

## **Screenshots/Recordings**

| Before       | After        |
|--------------|--------------|
| Screenshot 2024-04-18 at 3 56 43 PM |Screenshot 2024-04-18 at 3 56 43 PM |

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/UI/NetworkVerificationInfo/NetworkVerificationInfo.tsx [hunk 1]
```diff
@@ -256,10 +315,26 @@ const NetworkVerificationInfo = ({
...
-          <Text onPress={openHowToUseCustomNetworks}>
-            {strings('add_custom_network.warning_subtext_new.2')}
-          </Text>
+          {isMutichainVersion1Enabled ? (
+            <Text>
+              {strings(
+                'switch_custom_network.add_network_and_give_dapp_permission_warning',
+                {
+                  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                  // @ts-ignore
```
Comment (tommasini) [inline app/components/UI/NetworkVerificationInfo/NetworkVerificationInfo.tsx:324]: We should not use ts-ignore! If the error is expected, we can use ts-expect-error and write the reason of it

Comment (EtherWizard33) [reply app/components/UI/NetworkVerificationInfo/NetworkVerificationInfo.tsx]: good idea, thanks
done :)

**Review by mirceanis — CHANGES_REQUESTED** [review]
Looks good.
I think adding a test would help

**Comment by EtherWizard33** [comment]
> Looks good. I think adding a test would help

Hi Mircea, for this feature, like for the network UI redesign me and Salim from assets worked on last month, David Walsh wants us to write e2e and unit test at the end, before removing the the feature flag, this way the code is done and stable and any unecessary e2e tests and unit tests are removed/replaced at the same time. And the added advantage is that any existing unit tests, including snapshots, ensure that the UI when feature flag is off, does not change until we are ready to move from the old code to the new.

David is OOO this week and not there to explain this reasoning more in detail, but we did this for the network ui redesign also, still behind feature flag.

**Review by mirceanis — APPROVED** [review]
looks good

**Review by tommasini — APPROVED** [review]

**Pull request merged**